webapp/channels/src/components/new_search/search_hint.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import styled from 'styled-components';
+
+import {searchHintOptions, searchFilesHintOptions} from 'utils/constants';
+
+type Props = {
+    onSelectFilter: (filter: string) => void;
+    searchType: string;
+    searchTerms: string;
+    hasSelectedOption: boolean;
+    isDate: boolean;
+}
+
+const SearchHintsContainer = styled.div`
+    display: flex;
+    padding: 20px 24px;
+    color: rgba(var(--center-channel-color-rgb), 0.75);
+    i {
+        margin-right: 8px;
+        color: var(--center-channel-color-56);
+    }
+`;
+
+const SearchFilter = styled.button`
+    display: flex;
+    padding: 4px 10px;
+    color: var(--center-channel-color);
+    background: rgba(var(--center-channel-color-rgb), 0.08);
+    border-radius: var(--radius-l);
+    border: none;
+    font-size: 10px;
+    font-weight: 600;
+    line-height: 12px;
+    margin-left: 8px;
+    &:hover {
+        background: rgba(var(--center-channel-color-rgb), 0.16);
+    }
+`;
+
+const SearchHints = ({onSelectFilter, searchType, searchTerms, hasSelectedOption, isDate}: Props): JSX.Element => {
+    const intl = useIntl();
+    let filters = searchHintOptions.filter((filter) => filter.searchTerm !== '-' && filter.searchTerm !== '""');
+    if (searchType === 'files') {
+        filters = searchFilesHintOptions.filter((filter) => filter.searchTerm !== '-' && filter.searchTerm !== '""');
+    }
+
+    if (isDate) {
+        return <></>;
+    }
+
+    if (hasSelectedOption) {
+        return (
+            <SearchHintsContainer>
+                <i className='icon icon-keyboard-return'/>
+                <FormattedMessage
+                    id='search_hint.enter_to_select'
+                    defaultMessage='Press Enter to select'
+                />
+            </SearchHintsContainer>
+        );
+    }
+
+    if (searchTerms.length > 0 && searchTerms[searchTerms.length - 1] !== ' ') {
+        return (
+            <SearchHintsContainer>
+                <i className='icon icon-keyboard-return'/>
+                <FormattedMessage
+                    id='search_hint.enter_to_search'
+                    defaultMessage='Press Enter to search'
+                />
+            </SearchHintsContainer>
+        );
+    }
+
+    return (
+        <SearchHintsContainer>
+            <i className='icon icon-lightbulb-outline'/>
+            <FormattedMessage
+                id='search_hint.filter'
+                defaultMessage='Filter your search with:'
+            />
+            {filters.map((filter) => (
+                <SearchFilter
+                    key={filter.searchTerm}
+                    onClick={() => onSelectFilter(filter.searchTerm)}
```

Comment:
New function on every re-render.

Reply:
This one, like the other one, gets me into more complicated code and I don't want to do it until we see we need to gain that performance.